flexget/components/ftp/sftp_client.py

```diff
@@ -193,12 +193,17 @@ def path_exists(self, path: str) -> bool:
         """
         return self._sftp.lexists(path)
 
-    def make_dirs(self, path: str) -> None:
+    def make_dirs(self, path: str, limit=0) -> None:
         """
         Build directories
 
         :param path: path to build
         """
+
+        parent_path = Path(path).parent
+        if not self.path_exists(parent_path.as_posix()) and limit < 5:
+            self.make_dirs(parent_path.as_posix(), limit + 1)
```

Comment:
You should probably return an error if the max recursion depth is exceeded, right?